x-pack/platform/plugins/private/upgrade_assistant/server/lib/update_index/index.ts

```diff
@@ -26,6 +26,20 @@ export async function updateIndex({ esClient, index, operations }: UpdateIndexPa
 
     switch (operation) {
       case 'blockWrite': {
+        // hacky support for disabling rollup jobs before blocking write
+        const rollupCaps = await esClient.rollup.getRollupIndexCaps({ index });
```

Comment:
nit: perhaps this can be refactored so it can be used in both places

```typescript
async function handleRollupJob(esClient, index, action) {
  const validActions = ['unsetIfStopped', 'stopIfNotStopped'];
  if (!validActions.includes(action)) {
    throw new Error(`Invalid action: ${action}. Expected one of: ${validActions.join(', ')}`);
  }

  const rollupCaps = await esClient.rollup.getRollupIndexCaps({ index });
  ....
  const jobState = jobs.jobs[0].status.job_state;

  if (action === 'unsetIfStopped' && jobState === 'stopped') {
    return undefined;
  }

  if (action === 'stopIfNotStopped' && jobState !== 'stopped') {
    await esClient.rollup.stopJob({ id: rollupJob });
  }

  return rollupJob;
}
```